# TODO:
https://www.kaggle.com/competitions/playground-series-s3e2/discussion/378795
https://www.kaggle.com/competitions/playground-series-s3e2/discussion/378780

These show that we should incorporate original data but when validating using K fold methods, we should only validate based on data in the competition dataset and not on original dataset. So implement this technique for this competition

### Also, bagging resulted in a lot better score with simple mean in the last competition, thought it didn't score much better on public, it absolutely ranked up to 60th position in the final private LB. So,
## Remember to trust your cvs over pbl

# A Few more TODOs:
* select features with less than or equal to 10 values as catergorical features, instead of current 20, see if it improves the score
* Try target encoding, weights of evidence AND leave one out encoding, see which one performs better

# Imports

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import os
from pathlib import Path
import xgboost as xgb
import lightgbm as lgbm
import catboost
from sklearn.model_selection import StratifiedKFold, train_test_split
from sklearn.metrics import roc_auc_score
from IPython.display import display
from sklearn.preprocessing import LabelEncoder, OneHotEncoder, OrdinalEncoder
import optuna
from sklearn.preprocessing import StandardScaler

from category_encoders import TargetEncoder, LeaveOneOutEncoder, WOEEncoder

In [3]:
import warnings
warnings.filterwarnings('ignore')

# Loading Data

In [4]:
BASE_PATH = Path('../input/playground-series-s3e3')

# id is not going to be an informative feature, so we're dropping it for train
# but since we'll need test set's ids to make the submission file, so we'll save those in  a separate varible before dropping
train = pd.read_csv(BASE_PATH / "train.csv").drop(columns="id")
test = pd.read_csv(BASE_PATH / "test.csv")
test_idx = test.id
test = test.drop(columns="id")

# It's been shown that incorporating original data, improves scores - at least on the public leaderboard. So let's do that!
original = pd.read_csv('../input/ibm-hr-analytics-attrition-dataset/WA_Fn-UseC_-HR-Employee-Attrition.csv')

train.head()

Age     BusinessTravel  DailyRate              Department  \
0   36  Travel_Frequently        599  Research & Development   
1   35      Travel_Rarely        921                   Sales   
2   32      Travel_Rarely        718                   Sales   
3   38      Travel_Rarely       1488  Research & Development   
4   50      Travel_Rarely       1017  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  \
0                24          3        Medical              1   
1                 8          3          Other              1   
2                26          3      Marketing              1   
3                 2          3        Medical              1   
4                 5          4        Medical              1   

   EnvironmentSatisfaction  Gender  ...  StandardHours  StockOptionLevel  \
0                        4    Male  ...             80                 1   
1                        1    Male  ...             80                 1   
2                        3    Male  ...             80                 2   
3                        3  Female  ...             80                 0   
4                        2  Female  ...             80                 0   

   TotalWorkingYears TrainingTimesLastYear  WorkLifeBalance YearsAtCompany  \
0                 10                     2                3             10   
1                  4                     3                3              4   
2                  4                     3                3              3   
3                 15                     1                1              6   
4                 31                     0                3             31   

   YearsInCurrentRole  YearsSinceLastPromotion  YearsWithCurrManager Attrition  
0                   0                        7                     8         0  
1                   2                        0                     3         0  
2                   2                        1                     2         0  
3                   0                        0                     2         0  
4                  14                        4                    10         1  

[5 rows x 34 columns]

In [5]:
original.head()

Age Attrition     BusinessTravel  DailyRate              Department  \
0   41       Yes      Travel_Rarely       1102                   Sales   
1   49        No  Travel_Frequently        279  Research & Development   
2   37       Yes      Travel_Rarely       1373  Research & Development   
3   33        No  Travel_Frequently       1392  Research & Development   
4   27        No      Travel_Rarely        591  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeNumber  \
0                 1          2  Life Sciences              1               1   
1                 8          1  Life Sciences              1               2   
2                 2          2          Other              1               4   
3                 3          4  Life Sciences              1               5   
4                 2          1        Medical              1               7   

   ...  RelationshipSatisfaction StandardHours  StockOptionLevel  \
0  ...                         1            80                 0   
1  ...                         4            80                 1   
2  ...                         2            80                 0   
3  ...                         3            80                 0   
4  ...                         4            80                 1   

   TotalWorkingYears  TrainingTimesLastYear WorkLifeBalance  YearsAtCompany  \
0                  8                      0               1               6   
1                 10                      3               3              10   
2                  7                      3               3               0   
3                  8                      3               3               8   
4                  6                      3               3               2   

  YearsInCurrentRole  YearsSinceLastPromotion  YearsWithCurrManager  
0                  4                        0                     5  
1                  7                        1                     7  
2                  0                        0                     0  
3                  7                        3                     0  
4                  2                        2                     2  

[5 rows x 35 columns]

# Pre-Pre-Processing

### Let's make the feature names and order consistent b/w our competition dataset and original dataset, before we concatenate

In [6]:
original['Attrition'] = (original['Attrition'] == 'Yes').astype(np.int64)

# in original data, id is termed as "EmployeeNumber", so let's drop it
original.drop(columns="EmployeeNumber", inplace=True)

In [7]:
# now reordering the features in original dataset
original = original[list(train.columns)]

### Someone in the first compeition showed that adding a source feature i.e. a feature that indicates whether a given record is from original dataset or synthetic improves performance, so let's' do that!

In [8]:
original["is_original"] = 1
train["is_original"] = 0
test["is_original"] = 0

### Let's finally concatenate

In [9]:
train_extended = pd.concat([train, original]).reset_index(drop=True)
len(train_extended)

3147

### checking for null values

In [10]:
pd.concat([train_extended.isnull().sum().rename("Missing in Train"),
           test.isnull().sum().rename("Missing in Test")], axis=1).sort_values(by="Missing in Train")

Missing in Train  Missing in Test
Age                                      0              0.0
Over18                                   0              0.0
OverTime                                 0              0.0
PercentSalaryHike                        0              0.0
PerformanceRating                        0              0.0
RelationshipSatisfaction                 0              0.0
StandardHours                            0              0.0
NumCompaniesWorked                       0              0.0
StockOptionLevel                         0              0.0
TrainingTimesLastYear                    0              0.0
WorkLifeBalance                          0              0.0
YearsAtCompany                           0              0.0
YearsInCurrentRole                       0              0.0
YearsSinceLastPromotion                  0              0.0
YearsWithCurrManager                     0              0.0
TotalWorkingYears                        0              0.0
Attrition                                0              NaN
MonthlyRate                              0              0.0
MaritalStatus                            0              0.0
BusinessTravel                           0              0.0
DailyRate                                0              0.0
Department                               0              0.0
DistanceFromHome                         0              0.0
Education                                0              0.0
EducationField                           0              0.0
MonthlyIncome                            0              0.0
EmployeeCount                            0              0.0
Gender                                   0              0.0
HourlyRate                               0              0.0
JobInvolvement                           0              0.0
JobLevel                                 0              0.0
JobRole                                  0              0.0
JobSatisfaction                          0              0.0
EnvironmentSatisfaction                  0              0.0
is_original                              0              0.0

#### Insights: No missing values! Something to celebrate! :p

## Let's also concatenate test data to train

In [11]:
y = train_extended.Attrition
y

0       0
1       0
2       0
3       0
4       1
       ..
3142    0
3143    0
3144    0
3145    0
3146    0
Name: Attrition, Length: 3147, dtype: int64

In [12]:
df = pd.concat([train_extended.drop(columns="Attrition"), test])

# Preprocessing

### Identifying Categorical Features

In [13]:
df.dtypes.sort_values()

Age                          int64
YearsSinceLastPromotion      int64
YearsInCurrentRole           int64
YearsAtCompany               int64
WorkLifeBalance              int64
TrainingTimesLastYear        int64
TotalWorkingYears            int64
StockOptionLevel             int64
StandardHours                int64
RelationshipSatisfaction     int64
PerformanceRating            int64
PercentSalaryHike            int64
NumCompaniesWorked           int64
MonthlyRate                  int64
YearsWithCurrManager         int64
MonthlyIncome                int64
JobSatisfaction              int64
DailyRate                    int64
DistanceFromHome             int64
Education                    int64
EmployeeCount                int64
HourlyRate                   int64
EnvironmentSatisfaction      int64
JobLevel                     int64
JobInvolvement               int64
is_original                  int64
Gender                      object
MaritalStatus               object
OverTime            

### Remember, being of type int, doesn't mean that the feature cannot be categorial.
#### Let's check for unique values in each column

In [14]:
df.nunique().sort_values()

StandardHours                  1
EmployeeCount                  1
Over18                         1
is_original                    2
PerformanceRating              2
OverTime                       2
Gender                         2
BusinessTravel                 3
Department                     3
MaritalStatus                  3
RelationshipSatisfaction       4
JobSatisfaction                4
WorkLifeBalance                4
StockOptionLevel               5
JobInvolvement                 5
EnvironmentSatisfaction        5
Education                      6
JobLevel                       6
EducationField                 6
TrainingTimesLastYear          7
JobRole                        9
NumCompaniesWorked            11
PercentSalaryHike             15
YearsSinceLastPromotion       16
YearsWithCurrManager          18
YearsInCurrentRole            19
DistanceFromHome              29
YearsAtCompany                38
TotalWorkingYears             41
Age                           43
HourlyRate

#### INSIGHTS: Taking a quick look at number of unique values in features reveals that we should be safe setting the threshold for to 20 unique values for what consitutes as a categorical feature
#### We'll drop columns with only one value as they bring nothing to the table

#### But feel free to use your own intuition and test & trial to figure our what's works best in terms of threshold and features

In [15]:
feats_to_drop = [col for col in df.columns if df[col].nunique()==1]
cat_features = [col for col in df.columns if df[col].nunique() <= 20 and df[col].nunique() > 1]

In [16]:
df.drop(columns=feats_to_drop, inplace=True)

#### We won't use one hot encoder here, because we already have a large ratio of features to rows and one hotting would increase that ratio by a large margin even further which will result in severe overfitting
#### Rather we'll use ordinal/label encoder (they're basically the same thing)

In [17]:
len(df), len(y)

(4266, 3147)

In [18]:
# but first let's separate test and train_extended
X_train = df.iloc[:-len(test), :]
X_test = df.iloc[-len(test): , :]

In [19]:
target_enc = TargetEncoder()
loo_enc = LeaveOneOutEncoder(sigma=0.05)
woe_enc = WOEEncoder(sigma=0.05)

target_enc.fit(X_train[cat_features], y)

X_train[cat_features] = target_enc.transform(X_train[cat_features], y)
X_test[cat_features] = target_enc.transform(X_test[cat_features])

X_train.head()

Age  BusinessTravel  DailyRate  Department  DistanceFromHome  Education  \
0   36        0.223048        599    0.121711                24          3   
1   35        0.128590        921    0.173391                 8          3   
2   32        0.128590        718    0.173391                26          3   
3   38        0.128590       1488    0.121711                 2          3   
4   50        0.128590       1017    0.121711                 5          4   

   EducationField  EnvironmentSatisfaction    Gender  HourlyRate  ...  \
0        0.124383                        4  0.147996          42  ...   
1        0.103659                        1  0.147996          46  ...   
2        0.196141                        3  0.147996          80  ...   
3        0.124383                        3  0.124063          40  ...   
4        0.124383                        2  0.124063          37  ...   

   RelationshipSatisfaction  StockOptionLevel  TotalWorkingYears  \
0                         2                 1                 10   
1                         4                 1                  4   
2                         4                 2                  4   
3                         3                 0                 15   
4                         3                 0                 31   

   TrainingTimesLastYear  WorkLifeBalance  YearsAtCompany  YearsInCurrentRole  \
0                      2                3              10                   0   
1                      3                3               4                   2   
2                      3                3               3                   2   
3                      1                1               6                   0   
4                      0                3              31                  14   

   YearsSinceLastPromotion  YearsWithCurrManager  is_original  
0                        7                     8            0  
1                        0                     3            0  
2                        1                     2            0  
3                        0                     2            0  
4                        4                    10            0  

[5 rows x 31 columns]

In [20]:
# lets glue them back together
df = pd.concat([X_train, X_test])
len(df)

4266

## Always a good idea to scale the features

In [21]:
# we don't want to scale the source column
sc = StandardScaler()
df_scaled = df.copy()
df_scaled.iloc[:, :-1] = pd.DataFrame(sc.fit_transform(df.iloc[:, :-1]), columns=df.columns[:-1])
df_scaled.head()

Age  BusinessTravel  DailyRate  Department  DistanceFromHome  \
0 -0.042187        1.944557  -0.676304   -0.681781          1.914406   
1 -0.155733       -0.245174   0.160523    1.402552         -0.108063   
2 -0.496370       -0.245174  -0.367042    1.402552          2.167215   
3  0.184905       -0.245174   1.634065   -0.681781         -0.866489   
4  1.547454       -0.245174   0.410011   -0.681781         -0.487276   

   Education  EducationField  EnvironmentSatisfaction    Gender  HourlyRate  \
0   0.067799       -0.490679                 1.151976  0.786359   -1.276573   
1   0.067799       -1.189936                -1.590395  0.786359   -1.074371   
2   0.067799        1.930496                 0.237852  0.786359    0.644348   
3   0.067799       -0.490679                 0.237852 -1.271683   -1.377674   
4   1.041633       -0.490679                -0.676271 -1.271683   -1.529326   

   ...  RelationshipSatisfaction  StockOptionLevel  TotalWorkingYears  \
0  ...                 -0.680038          0.299291          -0.120890   
1  ...                  1.182590          0.299291          -0.927119   
2  ...                  1.182590          1.526962          -0.927119   
3  ...                  0.251276         -0.928379           0.550967   
4  ...                  0.251276         -0.928379           2.700910   

   TrainingTimesLastYear  WorkLifeBalance  YearsAtCompany  YearsInCurrentRole  \
0              -0.638312         0.351622        0.512223           -1.167739   
1               0.188354         0.351622       -0.489294           -0.609548   
2               0.188354         0.351622       -0.656214           -0.609548   
3              -1.464978        -2.630526       -0.155455           -1.167739   
4              -2.291644         0.351622        4.017536            2.739597   

   YearsSinceLastPromotion  YearsWithCurrManager  is_original  
0                 1.553534              1.052484            0  
1                -0.655171             -0.331151            0  
2                -0.339641             -0.607878            0  
3                -0.655171             -0.607878            0  
4                 0.606946              1.605938            0  

[5 rows x 31 columns]

### Let's seprate test and train sets

In [22]:
X_train = df_scaled.iloc[:-len(test), :]
X_test = df_scaled.iloc[-len(test): , :]

# Modelling

### But first, let's setup cross validation

In [23]:
# for i, (x, y) in enumerate(zip([1,2,3], [4,5,6])):
#     print(f"{'*'*10} {i}")
#     print(f"X: {x}")
#     print(f"Y: {y}")    

In [24]:
# a = np.array([1,2,3])
# b = np.array([4,5,6])

# np.append(a, b)

In [25]:
# we're gonna train on the combined dataset but, we'll only calculate the validation score only on comp data

# N_FOLDS = 10

def cross_validate(X, y, model, model_verbose=None, verbose=None):
    N_FOLDS = 5

    skf = StratifiedKFold(n_splits=N_FOLDS, shuffle=True, random_state=1337)
    skf_2 = StratifiedKFold(n_splits=N_FOLDS, shuffle=True, random_state=369)
    all_scores = []
    
    X_comp = X[X.is_original==0]
    y_comp = y[X_comp.index]
    X_original = X[X.is_original==1]
    y_original = y[X_original.index]
    
    # the k fold splitter created indcies based on the length from 0 to len-1, it doesn't take into account the index as I thought.
    # so we reset the index to match with the indices generated by splitter to access those elements
    X_original = X_original.reset_index(drop=True)
    y_original = y_original.reset_index(drop=True)
    
    for fold_id, ((train_comp_idx, val_comp_idx), (train_org_idx, val_org_idx)) \
                in enumerate(zip(skf.split(X_comp, y_comp), skf_2.split(X_original, y_original))):
        
        
        # for training we'll use data from both datasets
        X_tr_comp = X_comp.iloc[train_comp_idx]
        X_tr_org = X_original.iloc[train_org_idx]
        
        y_tr_comp = y_comp[train_comp_idx]
        y_tr_org = y_original[train_org_idx]
        
        
        # lets combine both
        X_tr = pd.concat([X_tr_comp, X_tr_org], axis=0).reset_index(drop=True)
        y_tr = pd.concat([y_tr_comp, y_tr_org], axis=0).reset_index(drop=True)
        
        
        # but for validation we'll use only competition data        
        X_val = X_comp.iloc[val_comp_idx]
        y_val = y_comp[val_comp_idx]
               
        model.fit(X_tr, y_tr, 
                  eval_set=[(X_val, y_val)],
                  early_stopping_rounds=50,
                 verbose=model_verbose)
        
        y_pred = model.predict_proba(X_val)[:, 1]
        
        auc = roc_auc_score(y_val, y_pred)
        
        print(f"Fold {fold_id} \t auc: {auc}")
        
        all_scores.append(auc)
    
    avg_auc = np.mean(all_scores)
    
    print(f"Avg AUC: {avg_auc}")

In [26]:
# random params values - make sure to tune yours
xgb_params = {'n_estimators': 150,
                 'max_depth': 3,
                 'learning_rate': 0.1,
                 'min_child_weight': 4,
                 'subsample': 0.7,
                 'colsample_bytree': 0.3
             }


xgb_clf = xgb.XGBClassifier(**xgb_params)

cross_validate(X_train, y, xgb_clf, model_verbose=False)

# xgb_clf.fit(X_train, y, verbose=0)

Fold 0 	 auc: 0.8373310810810811
Fold 1 	 auc: 0.9072635135135134
Fold 2 	 auc: 0.8171186440677966
Fold 3 	 auc: 0.8738983050847458
Fold 4 	 auc: 0.9008474576271186
Avg AUC: 0.867291800274851


In [ ]:
# random params but feel free to tune
lgbm_params = {'n_estimators': 407,
                 'num_rounds': 274,
                 'learning_rate': 0.1,
                 'num_leaves': 195,
                 'max_depth': 9,
                 'min_data_in_leaf': 46,
                 'lambda_l1': 0.01,
                 'lambda_l2': 0.6,
                 'min_gain_to_split': 1.42,
                 'bagging_fraction': 0.45,
                 'feature_fraction': 0.3}


lgbm_clf = lgbm.LGBMClassifier(**lgbm_params)

cross_validate(X_train, y, lgbm_clf, model_verbose=-1)

# lgbm_clf.fit(X_train, y, verbose=False)

In [50]:
#random params but feel free to tune
catboost_params = {'loss_function': 'CrossEntropy',
                     'learning_rate': 0.76,
                     'l2_leaf_reg': 0.014,
                     'colsample_bylevel': 0.06,
                     'depth': 1,
                     'boosting_type': 'Plain',
                     'bootstrap_type': 'Bernoulli',
                     'min_data_in_leaf': 18,
                     'one_hot_max_size': 14,
                     'subsample': 0.99}

catboost_clf = catboost.CatBoostClassifier(**catboost_params)

cross_validate(X_train, y, catboost_clf, model_verbose=False)

# catboost_clf.fit(X_train, y, verbose=False)

Fold 0 	 auc: 0.8514358108108109
Fold 1 	 auc: 0.9086993243243243
Fold 2 	 auc: 0.8044915254237287
Fold 3 	 auc: 0.8661864406779661
Fold 4 	 auc: 0.9154237288135594
Avg AUC: 0.8692473660100777


## INSIGHTS:
let's use this method of cross validation to
* Tune all our models
* Select top k
* Take their predictions average
* submit

In [75]:
np.random.randint(1, 10, size=(2,3))

array([[7, 2, 9],
       [7, 1, 7]])

In [83]:
some_X = pd.DataFrame(data=np.random.randint(1, 10, size=(2,3)))
pd.concat([some_X, some_X], axis=0).reset_index(drop=True)

0  1  2
0  6  1  9
1  3  3  5
2  6  1  9
3  3  3  5

# Hyperparameters Tuning

## XGBoost

In [31]:
def objective_xgb(trial, X, y):
    params = {
        "tree_method": "gpu_hist",
        'n_estimators': trial.suggest_int('n_estimators', 50, 500),
        'max_depth': trial.suggest_int('max_depth', 2, 10),
        'learning_rate': trial.suggest_loguniform('learning_rate', 0.01, 0.3),
        'min_child_weight': trial.suggest_int('min_child_weight', 1, 10),
        'gamma': trial.suggest_loguniform('gamma', 0.00001, 0.3),
        'subsample': trial.suggest_loguniform('subsample', 0.2, 1.0),
        'colsample_bytree': trial.suggest_loguniform('colsample_bytree', 0.2, 1.0),
        'reg_alpha': trial.suggest_loguniform('reg_alpha', 0.00001, 1.0),
        'reg_lambda': trial.suggest_loguniform('reg_lambda', 0.00001, 1.0),
        'early_stopping_rounds': trial.suggest_int("early_stoppig_rounds", 40, 100)
    }
    

    # we're gonna train on the combined dataset but, we'll only calculate the validation score only on comp data

    N_FOLDS = 5

    skf = StratifiedKFold(n_splits=N_FOLDS, shuffle=True, random_state=1337)
    skf_2 = StratifiedKFold(n_splits=N_FOLDS, shuffle=True, random_state=369)
    all_scores = []
    
    X_comp = X[X.is_original==0]
    y_comp = y[X_comp.index]
    X_original = X[X.is_original==1]
    y_original = y[X_original.index]
    
    # the k fold splitter created indcies based on the length from 0 to len-1, it doesn't take into account the index as I thought.
    # so we reset the index to match with the indices generated by splitter to access those elements
    X_original = X_original.reset_index(drop=True)
    y_original = y_original.reset_index(drop=True)
    
    for fold_id, ((train_comp_idx, val_comp_idx), (train_org_idx, val_org_idx)) \
                in enumerate(zip(skf.split(X_comp, y_comp), skf_2.split(X_original, y_original))):
        
        
        # for training we'll use data from both datasets
        X_tr_comp = X_comp.iloc[train_comp_idx]
        X_tr_org = X_original.iloc[train_org_idx]
        
        y_tr_comp = y_comp[train_comp_idx]
        y_tr_org = y_original[train_org_idx]
        
        
        # lets combine both
        X_tr = pd.concat([X_tr_comp, X_tr_org], axis=0).reset_index(drop=True)
        y_tr = pd.concat([y_tr_comp, y_tr_org], axis=0).reset_index(drop=True)
        
        
        # but for validation we'll use only competition data        
        X_val = X_comp.iloc[val_comp_idx]
        y_val = y_comp[val_comp_idx]
        
        model = xgb.XGBClassifier(**params)
        model.fit(X_tr, y_tr, eval_set=[(X_val, y_val)], verbose=False)
        
        y_pred = model.predict_proba(X_val)[:, 1]
        
        # for some reaason the score isn't even touch 90, but on pbL we're on 94, really weird. Let's check how much we score on original data
        
        auc = roc_auc_score(y_val, y_pred)
        
#         print(f"Fold {fold_id} \t auc: {auc}")
        
        all_scores.append(auc)
    
    avg_auc = np.mean(all_scores)
    
    print(f"Avg AUC: {avg_auc}")
    
    return avg_auc

In [32]:
study_xgb = optuna.create_study(study_name="xgboost_tuning", direction="maximize")
func = lambda trial: objective_xgb(trial, X_train, y)
study_xgb.optimize(func, n_trials=100)

[I 2023-01-21 07:19:16,616] A new study created in memory with name: xgboost_tuning
[I 2023-01-21 07:19:17,513] Trial 0 finished with value: 0.8514896358222629 and parameters: {'n_estimators': 292, 'max_depth': 7, 'learning_rate': 0.2599300901450483, 'min_child_weight': 7, 'gamma': 0.0006123196326410788, 'subsample': 0.5555352165233362, 'colsample_bytree': 0.5232475882135765, 'reg_alpha': 0.03767451568087707, 'reg_lambda': 6.591529774651591e-05, 'early_stoppig_rounds': 43}. Best is trial 0 with value: 0.8514896358222629.


Avg AUC: 0.8514896358222629


[I 2023-01-21 07:19:19,333] Trial 1 finished with value: 0.8618841044434264 and parameters: {'n_estimators': 437, 'max_depth': 4, 'learning_rate': 0.08480202209492158, 'min_child_weight': 7, 'gamma': 0.013796760877067646, 'subsample': 0.35234481267712675, 'colsample_bytree': 0.594228629283936, 'reg_alpha': 0.10673960618116766, 'reg_lambda': 1.1299768215765113e-05, 'early_stoppig_rounds': 77}. Best is trial 1 with value: 0.8618841044434264.


Avg AUC: 0.8618841044434264


[I 2023-01-21 07:19:21,629] Trial 2 finished with value: 0.8583121850664224 and parameters: {'n_estimators': 383, 'max_depth': 4, 'learning_rate': 0.04631304014448371, 'min_child_weight': 7, 'gamma': 0.1228425078609511, 'subsample': 0.3853975150208087, 'colsample_bytree': 0.21364246957667102, 'reg_alpha': 0.2735838325858485, 'reg_lambda': 0.4090330296468641, 'early_stoppig_rounds': 69}. Best is trial 1 with value: 0.8618841044434264.


Avg AUC: 0.8583121850664224


[I 2023-01-21 07:19:23,010] Trial 3 finished with value: 0.8484267063673844 and parameters: {'n_estimators': 427, 'max_depth': 10, 'learning_rate': 0.22825047561634987, 'min_child_weight': 7, 'gamma': 6.216811945322213e-05, 'subsample': 0.8727056032121777, 'colsample_bytree': 0.5072307203987192, 'reg_alpha': 0.40108338607214766, 'reg_lambda': 0.017603416339587816, 'early_stoppig_rounds': 47}. Best is trial 1 with value: 0.8618841044434264.


Avg AUC: 0.8484267063673844


[I 2023-01-21 07:19:24,519] Trial 4 finished with value: 0.8604897503435639 and parameters: {'n_estimators': 430, 'max_depth': 9, 'learning_rate': 0.13329961745589686, 'min_child_weight': 7, 'gamma': 0.0012287039650611786, 'subsample': 0.2737180676798829, 'colsample_bytree': 0.9322380501805135, 'reg_alpha': 0.017006360205285096, 'reg_lambda': 1.772139872047635e-05, 'early_stoppig_rounds': 96}. Best is trial 1 with value: 0.8618841044434264.


Avg AUC: 0.8604897503435639


[I 2023-01-21 07:19:28,289] Trial 5 finished with value: 0.8600624141090243 and parameters: {'n_estimators': 476, 'max_depth': 5, 'learning_rate': 0.030730139958466473, 'min_child_weight': 6, 'gamma': 0.0001939299442909087, 'subsample': 0.9707319736172808, 'colsample_bytree': 0.3877667246540289, 'reg_alpha': 0.0006669788448166403, 'reg_lambda': 2.0121493124322307e-05, 'early_stoppig_rounds': 53}. Best is trial 1 with value: 0.8618841044434264.


Avg AUC: 0.8600624141090243


[I 2023-01-21 07:19:29,149] Trial 6 finished with value: 0.8547758818140174 and parameters: {'n_estimators': 175, 'max_depth': 7, 'learning_rate': 0.28376615012258216, 'min_child_weight': 10, 'gamma': 0.011316658106284754, 'subsample': 0.2577370788157339, 'colsample_bytree': 0.9367390020235995, 'reg_alpha': 0.0017965639978765736, 'reg_lambda': 0.0010384634210668224, 'early_stoppig_rounds': 49}. Best is trial 1 with value: 0.8618841044434264.


Avg AUC: 0.8547758818140174


[I 2023-01-21 07:19:30,051] Trial 7 finished with value: 0.858251546037563 and parameters: {'n_estimators': 142, 'max_depth': 3, 'learning_rate': 0.29971274926111047, 'min_child_weight': 9, 'gamma': 2.3931590107375117e-05, 'subsample': 0.8018965479123832, 'colsample_bytree': 0.2254057471689599, 'reg_alpha': 0.001881628351154662, 'reg_lambda': 0.03682378314533954, 'early_stoppig_rounds': 80}. Best is trial 1 with value: 0.8618841044434264.


Avg AUC: 0.858251546037563


[I 2023-01-21 07:19:31,146] Trial 8 finished with value: 0.8485011452130096 and parameters: {'n_estimators': 122, 'max_depth': 4, 'learning_rate': 0.23162753874965342, 'min_child_weight': 1, 'gamma': 0.002027484220005672, 'subsample': 0.4493229039197007, 'colsample_bytree': 0.87853970027653, 'reg_alpha': 0.27557459864698214, 'reg_lambda': 5.570118605022863e-05, 'early_stoppig_rounds': 88}. Best is trial 1 with value: 0.8618841044434264.


Avg AUC: 0.8485011452130096


[I 2023-01-21 07:19:34,477] Trial 9 finished with value: 0.8644904947320201 and parameters: {'n_estimators': 497, 'max_depth': 3, 'learning_rate': 0.02260300519782697, 'min_child_weight': 3, 'gamma': 0.0011104495035911275, 'subsample': 0.2700744109732691, 'colsample_bytree': 0.6763306999154709, 'reg_alpha': 1.0086479528931865e-05, 'reg_lambda': 0.0002730377506517872, 'early_stoppig_rounds': 63}. Best is trial 9 with value: 0.8644904947320201.


Avg AUC: 0.8644904947320201


[I 2023-01-21 07:19:36,303] Trial 10 finished with value: 0.849910959688502 and parameters: {'n_estimators': 305, 'max_depth': 2, 'learning_rate': 0.015197846848972754, 'min_child_weight': 2, 'gamma': 0.2794945390230988, 'subsample': 0.2201546790850151, 'colsample_bytree': 0.3371122519937255, 'reg_alpha': 2.604598630387668e-05, 'reg_lambda': 0.0017192458874380474, 'early_stoppig_rounds': 62}. Best is trial 9 with value: 0.8644904947320201.


Avg AUC: 0.849910959688502


[I 2023-01-21 07:19:37,984] Trial 11 finished with value: 0.8706785387081999 and parameters: {'n_estimators': 485, 'max_depth': 2, 'learning_rate': 0.08350126827497835, 'min_child_weight': 4, 'gamma': 0.014707387142771743, 'subsample': 0.3378252488640548, 'colsample_bytree': 0.6696920481550677, 'reg_alpha': 1.6499846775461103e-05, 'reg_lambda': 0.00035530628629560387, 'early_stoppig_rounds': 76}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8706785387081999


[I 2023-01-21 07:19:41,128] Trial 12 finished with value: 0.8530718621163537 and parameters: {'n_estimators': 494, 'max_depth': 2, 'learning_rate': 0.010858209065780295, 'min_child_weight': 3, 'gamma': 0.010382287576959292, 'subsample': 0.307494564767379, 'colsample_bytree': 0.6781434440579159, 'reg_alpha': 1.0452686022317743e-05, 'reg_lambda': 0.00039877270872646026, 'early_stoppig_rounds': 63}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8530718621163537


[I 2023-01-21 07:19:41,910] Trial 13 finished with value: 0.8153384390746681 and parameters: {'n_estimators': 58, 'max_depth': 2, 'learning_rate': 0.025668682497882782, 'min_child_weight': 4, 'gamma': 0.037977926298856594, 'subsample': 0.5197850804714907, 'colsample_bytree': 0.7067441764569004, 'reg_alpha': 0.00015576251629793045, 'reg_lambda': 0.0002675475981768399, 'early_stoppig_rounds': 76}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8153384390746681


[I 2023-01-21 07:19:43,743] Trial 14 finished with value: 0.8667848144754924 and parameters: {'n_estimators': 372, 'max_depth': 6, 'learning_rate': 0.06905706239056235, 'min_child_weight': 4, 'gamma': 0.003590389121349993, 'subsample': 0.33110528226773717, 'colsample_bytree': 0.7098798601052304, 'reg_alpha': 8.252199362179719e-05, 'reg_lambda': 0.01256115091672632, 'early_stoppig_rounds': 59}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8667848144754924


[I 2023-01-21 07:19:45,867] Trial 15 finished with value: 0.8532986715529087 and parameters: {'n_estimators': 354, 'max_depth': 6, 'learning_rate': 0.08994677769304057, 'min_child_weight': 5, 'gamma': 0.006903733961698314, 'subsample': 0.6161963117944445, 'colsample_bytree': 0.4196140064467243, 'reg_alpha': 0.00010199529252198067, 'reg_lambda': 0.012660452697234741, 'early_stoppig_rounds': 55}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8532986715529087


[I 2023-01-21 07:19:47,875] Trial 16 finished with value: 0.8588780348144753 and parameters: {'n_estimators': 233, 'max_depth': 8, 'learning_rate': 0.05632968135264114, 'min_child_weight': 5, 'gamma': 0.0480737227656687, 'subsample': 0.20390664279190915, 'colsample_bytree': 0.7598096530686553, 'reg_alpha': 9.802666057303639e-05, 'reg_lambda': 0.08354112983896642, 'early_stoppig_rounds': 83}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8588780348144753


[I 2023-01-21 07:19:49,535] Trial 17 finished with value: 0.8567810352725607 and parameters: {'n_estimators': 361, 'max_depth': 6, 'learning_rate': 0.14116001245958373, 'min_child_weight': 4, 'gamma': 0.0035688906491622755, 'subsample': 0.36819031800193647, 'colsample_bytree': 0.28719482318279127, 'reg_alpha': 0.0004059519386796882, 'reg_lambda': 0.008263777424910105, 'early_stoppig_rounds': 93}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8567810352725607


[I 2023-01-21 07:19:54,017] Trial 18 finished with value: 0.8381085661933121 and parameters: {'n_estimators': 252, 'max_depth': 10, 'learning_rate': 0.056315748972101565, 'min_child_weight': 1, 'gamma': 0.0004006924066472754, 'subsample': 0.4270738288396976, 'colsample_bytree': 0.5461023483508652, 'reg_alpha': 3.604572248539308e-05, 'reg_lambda': 0.0037276162319212183, 'early_stoppig_rounds': 70}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8381085661933121


[I 2023-01-21 07:19:55,828] Trial 19 finished with value: 0.8601815162620247 and parameters: {'n_estimators': 397, 'max_depth': 5, 'learning_rate': 0.08323947743803957, 'min_child_weight': 3, 'gamma': 0.03311675362908905, 'subsample': 0.30079649115525753, 'colsample_bytree': 0.8185808764265085, 'reg_alpha': 0.00040407642984999345, 'reg_lambda': 0.17488853606989552, 'early_stoppig_rounds': 71}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8601815162620247


[I 2023-01-21 07:19:59,111] Trial 20 finished with value: 0.8605476408612001 and parameters: {'n_estimators': 337, 'max_depth': 7, 'learning_rate': 0.04071184629237528, 'min_child_weight': 4, 'gamma': 0.0035792146343874205, 'subsample': 0.33903396725262397, 'colsample_bytree': 0.46214853809058865, 'reg_alpha': 0.007488895325198282, 'reg_lambda': 0.0038876776767846457, 'early_stoppig_rounds': 60}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8605476408612001


[I 2023-01-21 07:20:02,263] Trial 21 finished with value: 0.8659281378836464 and parameters: {'n_estimators': 499, 'max_depth': 3, 'learning_rate': 0.021910497041951534, 'min_child_weight': 3, 'gamma': 0.0011860549550968364, 'subsample': 0.26801252831575517, 'colsample_bytree': 0.6587857570061662, 'reg_alpha': 1.3724346413870568e-05, 'reg_lambda': 0.0003900785327886439, 'early_stoppig_rounds': 66}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8659281378836464


[I 2023-01-21 07:20:03,521] Trial 22 finished with value: 0.8605837150710032 and parameters: {'n_estimators': 457, 'max_depth': 3, 'learning_rate': 0.10577472319909616, 'min_child_weight': 2, 'gamma': 0.00011646282742540673, 'subsample': 0.23585075750594514, 'colsample_bytree': 0.6420796116549333, 'reg_alpha': 3.850722496231393e-05, 'reg_lambda': 0.0007485274807785201, 'early_stoppig_rounds': 68}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8605837150710032


[I 2023-01-21 07:20:05,216] Trial 23 finished with value: 0.8677397503435639 and parameters: {'n_estimators': 416, 'max_depth': 5, 'learning_rate': 0.0649437183030122, 'min_child_weight': 5, 'gamma': 0.004116144195507306, 'subsample': 0.3056650076024149, 'colsample_bytree': 0.7915984599163035, 'reg_alpha': 1.8622790859900584e-05, 'reg_lambda': 6.415491570416832e-05, 'early_stoppig_rounds': 56}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8677397503435639


[I 2023-01-21 07:20:07,395] Trial 24 finished with value: 0.8510247938616583 and parameters: {'n_estimators': 400, 'max_depth': 6, 'learning_rate': 0.06729053115281397, 'min_child_weight': 5, 'gamma': 0.004593495516958626, 'subsample': 0.4418492250069204, 'colsample_bytree': 0.8258509904375447, 'reg_alpha': 6.196639796703332e-05, 'reg_lambda': 8.212545866751918e-05, 'early_stoppig_rounds': 56}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8510247938616583


[I 2023-01-21 07:20:08,654] Trial 25 finished with value: 0.8548471713238662 and parameters: {'n_estimators': 326, 'max_depth': 5, 'learning_rate': 0.15620947452575318, 'min_child_weight': 6, 'gamma': 0.02887698720568577, 'subsample': 0.319146263134395, 'colsample_bytree': 0.5887154391403232, 'reg_alpha': 0.0002022608276729304, 'reg_lambda': 0.00011129728724297497, 'early_stoppig_rounds': 58}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8548471713238662


[I 2023-01-21 07:20:11,972] Trial 26 finished with value: 0.8540779317453046 and parameters: {'n_estimators': 458, 'max_depth': 8, 'learning_rate': 0.039146997938361155, 'min_child_weight': 4, 'gamma': 0.08460650870222393, 'subsample': 0.41449974271913176, 'colsample_bytree': 0.9731269974712973, 'reg_alpha': 2.2363137619971018e-05, 'reg_lambda': 3.263099463009283e-05, 'early_stoppig_rounds': 41}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8540779317453046


[I 2023-01-21 07:20:13,747] Trial 27 finished with value: 0.8658733967017864 and parameters: {'n_estimators': 388, 'max_depth': 5, 'learning_rate': 0.11434862678868182, 'min_child_weight': 6, 'gamma': 0.00044532999123463686, 'subsample': 0.4936574555639689, 'colsample_bytree': 0.8160489581124437, 'reg_alpha': 4.055911451812812e-05, 'reg_lambda': 0.00014608356115714436, 'early_stoppig_rounds': 51}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8658733967017864


[I 2023-01-21 07:20:16,605] Trial 28 finished with value: 0.8555987173614292 and parameters: {'n_estimators': 423, 'max_depth': 8, 'learning_rate': 0.06586700923070454, 'min_child_weight': 2, 'gamma': 0.018034431749678396, 'subsample': 0.3764415718917662, 'colsample_bytree': 0.46684441502319285, 'reg_alpha': 0.0007021670616179983, 'reg_lambda': 0.0014289056127073621, 'early_stoppig_rounds': 73}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8555987173614292


[I 2023-01-21 07:20:17,861] Trial 29 finished with value: 0.8523814704535043 and parameters: {'n_estimators': 286, 'max_depth': 6, 'learning_rate': 0.17427570412618465, 'min_child_weight': 5, 'gamma': 0.002446423588551787, 'subsample': 0.6119044088987298, 'colsample_bytree': 0.7419924368586756, 'reg_alpha': 6.428653439288257e-05, 'reg_lambda': 0.03339416532086601, 'early_stoppig_rounds': 47}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8523814704535043


[I 2023-01-21 07:20:21,064] Trial 30 finished with value: 0.85796667430142 and parameters: {'n_estimators': 461, 'max_depth': 7, 'learning_rate': 0.03357924863158608, 'min_child_weight': 9, 'gamma': 0.007230578722867047, 'subsample': 0.2975686113594893, 'colsample_bytree': 0.5652294542026067, 'reg_alpha': 0.0002360208590945627, 'reg_lambda': 4.4828665817328306e-05, 'early_stoppig_rounds': 83}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.85796667430142


[I 2023-01-21 07:20:24,210] Trial 31 finished with value: 0.8650831424644985 and parameters: {'n_estimators': 477, 'max_depth': 3, 'learning_rate': 0.018700235520209822, 'min_child_weight': 3, 'gamma': 0.0015162568996944959, 'subsample': 0.23676107238241928, 'colsample_bytree': 0.6327534118944261, 'reg_alpha': 1.564473970325751e-05, 'reg_lambda': 0.0005311852293895938, 'early_stoppig_rounds': 66}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8650831424644985


[I 2023-01-21 07:20:25,915] Trial 32 finished with value: 0.8633234654145671 and parameters: {'n_estimators': 448, 'max_depth': 4, 'learning_rate': 0.06808723889235299, 'min_child_weight': 4, 'gamma': 0.0004887907359499556, 'subsample': 0.32838464135498485, 'colsample_bytree': 0.7563623966816564, 'reg_alpha': 1.806285061668625e-05, 'reg_lambda': 0.00014892145886881165, 'early_stoppig_rounds': 65}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8633234654145671


[I 2023-01-21 07:20:28,159] Trial 33 finished with value: 0.8687568140174072 and parameters: {'n_estimators': 413, 'max_depth': 2, 'learning_rate': 0.04689412880816038, 'min_child_weight': 3, 'gamma': 0.0007353800752411908, 'subsample': 0.2715044877680483, 'colsample_bytree': 0.5096718391175784, 'reg_alpha': 8.394818521814653e-05, 'reg_lambda': 0.002054970986653503, 'early_stoppig_rounds': 75}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8687568140174072


[I 2023-01-21 07:20:30,608] Trial 34 finished with value: 0.8663597114063215 and parameters: {'n_estimators': 416, 'max_depth': 2, 'learning_rate': 0.044666517281867144, 'min_child_weight': 5, 'gamma': 0.016589816026233026, 'subsample': 0.3386531163636669, 'colsample_bytree': 0.516870620000415, 'reg_alpha': 8.193842510296928e-05, 'reg_lambda': 0.0025039136449555963, 'early_stoppig_rounds': 75}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8663597114063215


[I 2023-01-21 07:20:32,303] Trial 35 finished with value: 0.857629409070087 and parameters: {'n_estimators': 370, 'max_depth': 4, 'learning_rate': 0.08777409482456275, 'min_child_weight': 2, 'gamma': 0.000228472453398413, 'subsample': 0.28654938787367634, 'colsample_bytree': 0.49330893622491256, 'reg_alpha': 0.08288681271407086, 'reg_lambda': 0.0052664877967020934, 'early_stoppig_rounds': 79}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.857629409070087


[I 2023-01-21 07:20:33,963] Trial 36 finished with value: 0.8690976866697205 and parameters: {'n_estimators': 415, 'max_depth': 2, 'learning_rate': 0.07334740797789553, 'min_child_weight': 4, 'gamma': 0.0007086980868527834, 'subsample': 0.24145686363330587, 'colsample_bytree': 0.6124933636910038, 'reg_alpha': 3.744063825020901e-05, 'reg_lambda': 0.026652592144683837, 'early_stoppig_rounds': 44}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8690976866697205


[I 2023-01-21 07:20:35,657] Trial 37 finished with value: 0.8604592876775079 and parameters: {'n_estimators': 417, 'max_depth': 2, 'learning_rate': 0.05189146651358283, 'min_child_weight': 6, 'gamma': 0.000836162386597265, 'subsample': 0.24754461559357935, 'colsample_bytree': 0.5719917763553489, 'reg_alpha': 2.3533970302829298e-05, 'reg_lambda': 1.1067251531325596e-05, 'early_stoppig_rounds': 44}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8604592876775079


[I 2023-01-21 07:20:37,025] Trial 38 finished with value: 0.8580495877233165 and parameters: {'n_estimators': 438, 'max_depth': 2, 'learning_rate': 0.10554010337242468, 'min_child_weight': 8, 'gamma': 7.157858818229888e-05, 'subsample': 0.21676689338168859, 'colsample_bytree': 0.4059563398267135, 'reg_alpha': 4.360512490035926e-05, 'reg_lambda': 0.46729760257373293, 'early_stoppig_rounds': 45}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8580495877233165


[I 2023-01-21 07:20:39,297] Trial 39 finished with value: 0.8586918231791113 and parameters: {'n_estimators': 317, 'max_depth': 4, 'learning_rate': 0.032504280259919506, 'min_child_weight': 3, 'gamma': 2.6939788459413902e-05, 'subsample': 0.3935589184985205, 'colsample_bytree': 0.34776590575233607, 'reg_alpha': 0.6565409630048534, 'reg_lambda': 0.0328823840948411, 'early_stoppig_rounds': 40}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8586918231791113


[I 2023-01-21 07:20:41,445] Trial 40 finished with value: 0.8614077530920751 and parameters: {'n_estimators': 407, 'max_depth': 3, 'learning_rate': 0.04942619385010267, 'min_child_weight': 7, 'gamma': 1.0829665099077925e-05, 'subsample': 0.2527077002378074, 'colsample_bytree': 0.5936385118687763, 'reg_alpha': 0.023644018273477507, 'reg_lambda': 0.08305523274438763, 'early_stoppig_rounds': 83}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8614077530920751


[I 2023-01-21 07:20:42,785] Trial 41 finished with value: 0.8650230187814933 and parameters: {'n_estimators': 377, 'max_depth': 2, 'learning_rate': 0.07890769117266391, 'min_child_weight': 4, 'gamma': 0.0021024521196391943, 'subsample': 0.2773852817006701, 'colsample_bytree': 0.6297878978691118, 'reg_alpha': 0.00015883283531224507, 'reg_lambda': 0.016983354533526458, 'early_stoppig_rounds': 52}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8650230187814933


[I 2023-01-21 07:20:44,894] Trial 42 finished with value: 0.8632963811268896 and parameters: {'n_estimators': 349, 'max_depth': 3, 'learning_rate': 0.06580614591990008, 'min_child_weight': 4, 'gamma': 0.0007816550066341677, 'subsample': 0.3646267142534527, 'colsample_bytree': 0.7203219840635603, 'reg_alpha': 0.0003183893132016129, 'reg_lambda': 0.010426412737700964, 'early_stoppig_rounds': 73}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8632963811268896


[I 2023-01-21 07:20:46,966] Trial 43 finished with value: 0.8599327187356849 and parameters: {'n_estimators': 472, 'max_depth': 5, 'learning_rate': 0.0729871436909579, 'min_child_weight': 5, 'gamma': 0.006014072979341872, 'subsample': 0.23161327249906236, 'colsample_bytree': 0.9064512062811733, 'reg_alpha': 5.486643865356437e-05, 'reg_lambda': 0.0021657221264813776, 'early_stoppig_rounds': 88}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8599327187356849


[I 2023-01-21 07:20:48,810] Trial 44 finished with value: 0.8672886509390747 and parameters: {'n_estimators': 444, 'max_depth': 2, 'learning_rate': 0.05789537386904777, 'min_child_weight': 3, 'gamma': 0.00030838513856859133, 'subsample': 0.2886321009061691, 'colsample_bytree': 0.5134727466008255, 'reg_alpha': 0.0019328358024570313, 'reg_lambda': 0.937979940118997, 'early_stoppig_rounds': 50}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8672886509390747


[I 2023-01-21 07:20:50,607] Trial 45 finished with value: 0.867879523591388 and parameters: {'n_estimators': 436, 'max_depth': 2, 'learning_rate': 0.05929233695151679, 'min_child_weight': 3, 'gamma': 0.0002536943785613681, 'subsample': 0.25967775349252836, 'colsample_bytree': 0.5000827123490448, 'reg_alpha': 0.0008328640274042373, 'reg_lambda': 0.9990870519215723, 'early_stoppig_rounds': 51}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.867879523591388


[I 2023-01-21 07:20:52,030] Trial 46 finished with value: 0.8658606848373799 and parameters: {'n_estimators': 483, 'max_depth': 3, 'learning_rate': 0.09704622396242744, 'min_child_weight': 2, 'gamma': 0.00016296683669019593, 'subsample': 0.20027536984986571, 'colsample_bytree': 0.44106369729741735, 'reg_alpha': 0.003936803377982773, 'reg_lambda': 0.0009615690796803007, 'early_stoppig_rounds': 47}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8658606848373799


[I 2023-01-21 07:20:53,252] Trial 47 finished with value: 0.8668270728355474 and parameters: {'n_estimators': 436, 'max_depth': 2, 'learning_rate': 0.1216566792628671, 'min_child_weight': 1, 'gamma': 0.00010014795608617169, 'subsample': 0.2641542048595137, 'colsample_bytree': 0.36555956526636013, 'reg_alpha': 1.065590717942179e-05, 'reg_lambda': 2.6438007046536193e-05, 'early_stoppig_rounds': 54}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8668270728355474


[I 2023-01-21 07:20:54,667] Trial 48 finished with value: 0.8611504237288136 and parameters: {'n_estimators': 195, 'max_depth': 3, 'learning_rate': 0.04086588811490805, 'min_child_weight': 3, 'gamma': 0.0005910569614093336, 'subsample': 0.22166806030713782, 'colsample_bytree': 0.486788078653009, 'reg_alpha': 0.0008829878489255849, 'reg_lambda': 0.00023616124764354755, 'early_stoppig_rounds': 100}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8611504237288136


[I 2023-01-21 07:20:57,693] Trial 49 finished with value: 0.8639558520384792 and parameters: {'n_estimators': 395, 'max_depth': 4, 'learning_rate': 0.028160523415627612, 'min_child_weight': 4, 'gamma': 0.00031217146218778135, 'subsample': 0.3062659819425908, 'colsample_bytree': 0.5376540599234703, 'reg_alpha': 0.00011737034081510113, 'reg_lambda': 0.07777816884732552, 'early_stoppig_rounds': 80}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8639558520384792


[I 2023-01-21 07:20:59,760] Trial 50 finished with value: 0.8617866468163078 and parameters: {'n_estimators': 432, 'max_depth': 2, 'learning_rate': 0.04740232826268315, 'min_child_weight': 5, 'gamma': 0.010945771802705806, 'subsample': 0.2522830410113598, 'colsample_bytree': 0.23997155512396737, 'reg_alpha': 2.8087826718128983e-05, 'reg_lambda': 0.17197207202691275, 'early_stoppig_rounds': 42}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8617866468163078


[I 2023-01-21 07:21:01,633] Trial 51 finished with value: 0.8672049358680715 and parameters: {'n_estimators': 443, 'max_depth': 2, 'learning_rate': 0.056560908694559306, 'min_child_weight': 3, 'gamma': 0.0002784108077180663, 'subsample': 0.2857243053403096, 'colsample_bytree': 0.5223933137909245, 'reg_alpha': 0.0023030060004133115, 'reg_lambda': 0.7244630911144476, 'early_stoppig_rounds': 49}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8672049358680715


[I 2023-01-21 07:21:03,561] Trial 52 finished with value: 0.8646446976637655 and parameters: {'n_estimators': 463, 'max_depth': 2, 'learning_rate': 0.06212092876144165, 'min_child_weight': 3, 'gamma': 0.0007608049499260153, 'subsample': 0.283002256898358, 'colsample_bytree': 0.4345931597718186, 'reg_alpha': 0.006152183656596544, 'reg_lambda': 0.8378197088301922, 'early_stoppig_rounds': 50}. Best is trial 11 with value: 0.8706785387081999.


Avg AUC: 0.8646446976637655


[I 2023-01-21 07:21:05,218] Trial 53 finished with value: 0.8725910444342648 and parameters: {'n_estimators': 486, 'max_depth': 2, 'learning_rate': 0.07874886385356504, 'min_child_weight': 2, 'gamma': 0.00017917494217713564, 'subsample': 0.31436946208977806, 'colsample_bytree': 0.6098822240806442, 'reg_alpha': 0.0013230495142811719, 'reg_lambda': 0.22601446554253807, 'early_stoppig_rounds': 45}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8725910444342648


[I 2023-01-21 07:21:06,573] Trial 54 finished with value: 0.863771300961979 and parameters: {'n_estimators': 488, 'max_depth': 3, 'learning_rate': 0.07773840766437659, 'min_child_weight': 2, 'gamma': 4.7706715133984684e-05, 'subsample': 0.3505140828098993, 'colsample_bytree': 0.6145104887730163, 'reg_alpha': 0.013142930421978834, 'reg_lambda': 0.3253148218080058, 'early_stoppig_rounds': 43}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.863771300961979


[I 2023-01-21 07:21:07,929] Trial 55 finished with value: 0.86488215758131 and parameters: {'n_estimators': 500, 'max_depth': 2, 'learning_rate': 0.08689726501661338, 'min_child_weight': 1, 'gamma': 0.00012131408847225708, 'subsample': 0.3184560420279638, 'colsample_bytree': 0.7969481405121284, 'reg_alpha': 0.0010544904861974794, 'reg_lambda': 0.19942481752837674, 'early_stoppig_rounds': 46}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.86488215758131


[I 2023-01-21 07:21:08,451] Trial 56 finished with value: 0.8405018323408153 and parameters: {'n_estimators': 57, 'max_depth': 3, 'learning_rate': 0.03618012470556812, 'min_child_weight': 2, 'gamma': 0.0012270115070198997, 'subsample': 0.2677931041854858, 'colsample_bytree': 0.680826437252808, 'reg_alpha': 0.0004134597636902784, 'reg_lambda': 0.04968129659954235, 'early_stoppig_rounds': 56}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8405018323408153


[I 2023-01-21 07:21:09,765] Trial 57 finished with value: 0.8583358909757216 and parameters: {'n_estimators': 469, 'max_depth': 9, 'learning_rate': 0.09789589335517676, 'min_child_weight': 4, 'gamma': 0.0001818617984804017, 'subsample': 0.21544445290519812, 'colsample_bytree': 0.8664750958937614, 'reg_alpha': 2.9678905220462868e-05, 'reg_lambda': 0.32729341610289076, 'early_stoppig_rounds': 48}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8583358909757216


[I 2023-01-21 07:21:11,196] Trial 58 finished with value: 0.865915597801191 and parameters: {'n_estimators': 411, 'max_depth': 2, 'learning_rate': 0.12635756147509034, 'min_child_weight': 3, 'gamma': 0.002562318720440974, 'subsample': 0.2428735548661282, 'colsample_bytree': 0.5571979140478335, 'reg_alpha': 0.0011423326348731606, 'reg_lambda': 0.006309894986147447, 'early_stoppig_rounds': 78}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.865915597801191


[I 2023-01-21 07:21:12,141] Trial 59 finished with value: 0.8502541227668348 and parameters: {'n_estimators': 119, 'max_depth': 3, 'learning_rate': 0.04514106509226612, 'min_child_weight': 4, 'gamma': 4.3105055351648105e-05, 'subsample': 0.7947772937959197, 'colsample_bytree': 0.6063378929394655, 'reg_alpha': 1.3728563056806042e-05, 'reg_lambda': 0.12634202717535703, 'early_stoppig_rounds': 53}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8502541227668348


[I 2023-01-21 07:21:13,951] Trial 60 finished with value: 0.861676420064132 and parameters: {'n_estimators': 382, 'max_depth': 4, 'learning_rate': 0.07573195068796365, 'min_child_weight': 2, 'gamma': 0.021898968279959952, 'subsample': 0.39882817866318454, 'colsample_bytree': 0.6892044979704965, 'reg_alpha': 0.0005687348521404854, 'reg_lambda': 0.489306570708623, 'early_stoppig_rounds': 60}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.861676420064132


[I 2023-01-21 07:21:15,767] Trial 61 finished with value: 0.8633311383417317 and parameters: {'n_estimators': 449, 'max_depth': 2, 'learning_rate': 0.055172390066039995, 'min_child_weight': 3, 'gamma': 0.0003098890884736869, 'subsample': 0.2985664227493318, 'colsample_bytree': 0.48215038887401185, 'reg_alpha': 0.0032553617802002907, 'reg_lambda': 0.7848224150709462, 'early_stoppig_rounds': 51}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8633311383417317


[I 2023-01-21 07:21:17,919] Trial 62 finished with value: 0.8652651740723775 and parameters: {'n_estimators': 426, 'max_depth': 2, 'learning_rate': 0.06128218578529596, 'min_child_weight': 3, 'gamma': 0.0003936355531931752, 'subsample': 0.3555856925690445, 'colsample_bytree': 0.5113731163985188, 'reg_alpha': 0.0016915694855255288, 'reg_lambda': 0.9192611498403387, 'early_stoppig_rounds': 45}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8652651740723775


[I 2023-01-21 07:21:20,091] Trial 63 finished with value: 0.8706899335776456 and parameters: {'n_estimators': 479, 'max_depth': 2, 'learning_rate': 0.05674878807656348, 'min_child_weight': 4, 'gamma': 0.0009567429126308685, 'subsample': 0.25944602995436694, 'colsample_bytree': 0.6461806245149115, 'reg_alpha': 0.0045200711573723535, 'reg_lambda': 0.25218803646114984, 'early_stoppig_rounds': 58}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8706899335776456


[I 2023-01-21 07:21:22,317] Trial 64 finished with value: 0.8641889028859367 and parameters: {'n_estimators': 480, 'max_depth': 3, 'learning_rate': 0.04901650915104436, 'min_child_weight': 5, 'gamma': 0.000903035827304304, 'subsample': 0.2646769192075499, 'colsample_bytree': 0.6574055696741062, 'reg_alpha': 0.005634363388609368, 'reg_lambda': 0.25573287086101126, 'early_stoppig_rounds': 58}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8641889028859367


[I 2023-01-21 07:21:24,281] Trial 65 finished with value: 0.8678275309207513 and parameters: {'n_estimators': 488, 'max_depth': 2, 'learning_rate': 0.07247697756765228, 'min_child_weight': 4, 'gamma': 0.0014598120244502784, 'subsample': 0.3200625470963042, 'colsample_bytree': 0.7807197760705806, 'reg_alpha': 0.014870644821296807, 'reg_lambda': 0.12156167010821924, 'early_stoppig_rounds': 62}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8678275309207513


[I 2023-01-21 07:21:25,333] Trial 66 finished with value: 0.8669731447549245 and parameters: {'n_estimators': 491, 'max_depth': 2, 'learning_rate': 0.14556185904212668, 'min_child_weight': 4, 'gamma': 0.18693666396446604, 'subsample': 0.2332104599251157, 'colsample_bytree': 0.7342467079586071, 'reg_alpha': 0.013868262974775277, 'reg_lambda': 0.4342361767874546, 'early_stoppig_rounds': 68}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8669731447549245


[I 2023-01-21 07:21:26,772] Trial 67 finished with value: 0.8651270613834173 and parameters: {'n_estimators': 457, 'max_depth': 2, 'learning_rate': 0.092473046179866, 'min_child_weight': 4, 'gamma': 0.0015306435434426868, 'subsample': 0.311624484739847, 'colsample_bytree': 0.5816749767787587, 'reg_alpha': 0.04673913952357615, 'reg_lambda': 0.13309526916519235, 'early_stoppig_rounds': 61}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8651270613834173


[I 2023-01-21 07:21:28,140] Trial 68 finished with value: 0.8574211520842878 and parameters: {'n_estimators': 473, 'max_depth': 3, 'learning_rate': 0.07140344063705573, 'min_child_weight': 2, 'gamma': 0.0004950231414425321, 'subsample': 0.25494080923123247, 'colsample_bytree': 0.6491853786638576, 'reg_alpha': 0.009227382734206556, 'reg_lambda': 0.02012814412768236, 'early_stoppig_rounds': 73}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8574211520842878


[I 2023-01-21 07:21:29,613] Trial 69 finished with value: 0.8680012024736602 and parameters: {'n_estimators': 488, 'max_depth': 2, 'learning_rate': 0.10898463281147651, 'min_child_weight': 3, 'gamma': 0.0005817126093197399, 'subsample': 0.3260087461288582, 'colsample_bytree': 0.7719005834514366, 'reg_alpha': 0.03366454496644411, 'reg_lambda': 0.06123033097906576, 'early_stoppig_rounds': 62}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8680012024736602


[I 2023-01-21 07:21:30,853] Trial 70 finished with value: 0.8629731447549244 and parameters: {'n_estimators': 455, 'max_depth': 2, 'learning_rate': 0.21066268333063198, 'min_child_weight': 3, 'gamma': 0.0006191427621954806, 'subsample': 0.46788278505228315, 'colsample_bytree': 0.54742779581971, 'reg_alpha': 0.15894674230064978, 'reg_lambda': 0.06448057376396424, 'early_stoppig_rounds': 64}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8629731447549244


[I 2023-01-21 07:21:32,753] Trial 71 finished with value: 0.8643754008245533 and parameters: {'n_estimators': 485, 'max_depth': 2, 'learning_rate': 0.08522376555087569, 'min_child_weight': 4, 'gamma': 0.0014276137879899282, 'subsample': 0.3473329361392967, 'colsample_bytree': 0.8466610632439675, 'reg_alpha': 0.0359009812723257, 'reg_lambda': 0.1286452003973912, 'early_stoppig_rounds': 63}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8643754008245533


[I 2023-01-21 07:21:34,146] Trial 72 finished with value: 0.8571606733852496 and parameters: {'n_estimators': 469, 'max_depth': 3, 'learning_rate': 0.09793936466936759, 'min_child_weight': 2, 'gamma': 0.0010116912815615314, 'subsample': 0.33197068119546946, 'colsample_bytree': 0.77757067952193, 'reg_alpha': 0.023915397593970246, 'reg_lambda': 0.03993780166126799, 'early_stoppig_rounds': 70}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8571606733852496


[I 2023-01-21 07:21:35,703] Trial 73 finished with value: 0.8648179111314704 and parameters: {'n_estimators': 262, 'max_depth': 2, 'learning_rate': 0.11081915533608007, 'min_child_weight': 3, 'gamma': 0.0028635128878610953, 'subsample': 0.2743334529879152, 'colsample_bytree': 0.6973888845910652, 'reg_alpha': 0.061127033395549216, 'reg_lambda': 0.024616561570947597, 'early_stoppig_rounds': 67}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8648179111314704


[I 2023-01-21 07:21:37,469] Trial 74 finished with value: 0.8714125057260651 and parameters: {'n_estimators': 495, 'max_depth': 2, 'learning_rate': 0.07722737720884305, 'min_child_weight': 4, 'gamma': 0.0019513955607966924, 'subsample': 0.3235425947810543, 'colsample_bytree': 0.7563892563002135, 'reg_alpha': 0.003782377290078836, 'reg_lambda': 0.5738062006098661, 'early_stoppig_rounds': 75}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8714125057260651


[I 2023-01-21 07:21:39,139] Trial 75 finished with value: 0.859864521300962 and parameters: {'n_estimators': 497, 'max_depth': 2, 'learning_rate': 0.08166037135902203, 'min_child_weight': 3, 'gamma': 0.0006178585148619767, 'subsample': 0.37830238419353246, 'colsample_bytree': 0.46206919898897175, 'reg_alpha': 0.0014952746006774733, 'reg_lambda': 0.6384127690598823, 'early_stoppig_rounds': 74}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.859864521300962


[I 2023-01-21 07:21:42,592] Trial 76 finished with value: 0.8575632730187814 and parameters: {'n_estimators': 500, 'max_depth': 3, 'learning_rate': 0.010395366953183483, 'min_child_weight': 4, 'gamma': 0.0002226234715029724, 'subsample': 0.29283136316450503, 'colsample_bytree': 0.6161954374735248, 'reg_alpha': 0.003962243972250703, 'reg_lambda': 0.5594802591962977, 'early_stoppig_rounds': 76}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8575632730187814


[I 2023-01-21 07:21:44,759] Trial 77 finished with value: 0.8602741639945031 and parameters: {'n_estimators': 476, 'max_depth': 2, 'learning_rate': 0.04239937379373405, 'min_child_weight': 5, 'gamma': 0.05462953081430302, 'subsample': 0.21039094910070705, 'colsample_bytree': 0.722407915191886, 'reg_alpha': 0.002810362808706811, 'reg_lambda': 0.30495496927261745, 'early_stoppig_rounds': 81}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8602741639945031


[I 2023-01-21 07:21:47,365] Trial 78 finished with value: 0.8648340586349061 and parameters: {'n_estimators': 431, 'max_depth': 2, 'learning_rate': 0.05207065919040741, 'min_child_weight': 1, 'gamma': 0.0019037785274045765, 'subsample': 0.24060281075123752, 'colsample_bytree': 0.6683078160017745, 'reg_alpha': 0.0005525260169523289, 'reg_lambda': 0.22139845812990355, 'early_stoppig_rounds': 76}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8648340586349061


[I 2023-01-21 07:21:49,488] Trial 79 finished with value: 0.8584040311497938 and parameters: {'n_estimators': 449, 'max_depth': 3, 'learning_rate': 0.060405660226122836, 'min_child_weight': 10, 'gamma': 0.00014354130265077612, 'subsample': 0.227320207990252, 'colsample_bytree': 0.9395808664802344, 'reg_alpha': 0.00025657589521890155, 'reg_lambda': 0.3371210908844503, 'early_stoppig_rounds': 71}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8584040311497938


[I 2023-01-21 07:21:52,167] Trial 80 finished with value: 0.8685404260192395 and parameters: {'n_estimators': 462, 'max_depth': 2, 'learning_rate': 0.0363723231652233, 'min_child_weight': 2, 'gamma': 9.0657602298059e-05, 'subsample': 0.2580219888408366, 'colsample_bytree': 0.6329153834451964, 'reg_alpha': 0.16228403657533894, 'reg_lambda': 0.06222803399102884, 'early_stoppig_rounds': 87}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8685404260192395


[I 2023-01-21 07:21:54,819] Trial 81 finished with value: 0.8643717934035731 and parameters: {'n_estimators': 463, 'max_depth': 2, 'learning_rate': 0.0355141291840676, 'min_child_weight': 2, 'gamma': 8.033669727171328e-05, 'subsample': 0.25787198634323266, 'colsample_bytree': 0.5848205657771238, 'reg_alpha': 0.35675988152973487, 'reg_lambda': 0.055802242931867634, 'early_stoppig_rounds': 86}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8643717934035731


[I 2023-01-21 07:21:57,367] Trial 82 finished with value: 0.8685700870361888 and parameters: {'n_estimators': 403, 'max_depth': 2, 'learning_rate': 0.037602372876945324, 'min_child_weight': 2, 'gamma': 0.00037848366126097234, 'subsample': 0.24535100817469635, 'colsample_bytree': 0.6359112116295019, 'reg_alpha': 0.009292334816360227, 'reg_lambda': 0.11001083636221821, 'early_stoppig_rounds': 95}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8685700870361888


[I 2023-01-21 07:22:00,082] Trial 83 finished with value: 0.8660317796610169 and parameters: {'n_estimators': 403, 'max_depth': 3, 'learning_rate': 0.02723024904986893, 'min_child_weight': 1, 'gamma': 0.0004790952480801106, 'subsample': 0.2454732456102873, 'colsample_bytree': 0.6359461394557208, 'reg_alpha': 0.1432024215785841, 'reg_lambda': 0.09298064777324289, 'early_stoppig_rounds': 94}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8660317796610169


[I 2023-01-21 07:22:02,913] Trial 84 finished with value: 0.8661809436555199 and parameters: {'n_estimators': 481, 'max_depth': 2, 'learning_rate': 0.031554728751826, 'min_child_weight': 2, 'gamma': 9.69240291444092e-05, 'subsample': 0.2763279061089855, 'colsample_bytree': 0.7069060225474734, 'reg_alpha': 0.01006098716333859, 'reg_lambda': 0.09492340025900241, 'early_stoppig_rounds': 93}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8661809436555199


[I 2023-01-21 07:22:05,002] Trial 85 finished with value: 0.8660841731562071 and parameters: {'n_estimators': 365, 'max_depth': 2, 'learning_rate': 0.03975303059303769, 'min_child_weight': 2, 'gamma': 0.0006754356973754166, 'subsample': 0.3244465413134134, 'colsample_bytree': 0.7538820094063865, 'reg_alpha': 0.023646640866867942, 'reg_lambda': 0.04360532676406327, 'early_stoppig_rounds': 89}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8660841731562071


[I 2023-01-21 07:22:06,551] Trial 86 finished with value: 0.8616702359138799 and parameters: {'n_estimators': 420, 'max_depth': 2, 'learning_rate': 0.10282410832208887, 'min_child_weight': 1, 'gamma': 0.0010741112137757113, 'subsample': 0.3057395612076254, 'colsample_bytree': 0.6565406059678575, 'reg_alpha': 0.004424293532377458, 'reg_lambda': 0.15089843976156828, 'early_stoppig_rounds': 99}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8616702359138799


[I 2023-01-21 07:22:09,366] Trial 87 finished with value: 0.8624942739349519 and parameters: {'n_estimators': 393, 'max_depth': 3, 'learning_rate': 0.024003004735192453, 'min_child_weight': 2, 'gamma': 0.0004071428252068029, 'subsample': 0.22927181970602656, 'colsample_bytree': 0.625842021789928, 'reg_alpha': 0.00891487977397198, 'reg_lambda': 0.028578408843471945, 'early_stoppig_rounds': 85}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8624942739349519


[I 2023-01-21 07:22:12,133] Trial 88 finished with value: 0.8638006184150251 and parameters: {'n_estimators': 468, 'max_depth': 2, 'learning_rate': 0.03655422609635484, 'min_child_weight': 3, 'gamma': 0.000187178908711561, 'subsample': 0.27920478333256377, 'colsample_bytree': 0.5998723053823739, 'reg_alpha': 0.6969670086391375, 'reg_lambda': 0.0006547876649453756, 'early_stoppig_rounds': 97}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8638006184150251


[I 2023-01-21 07:22:14,777] Trial 89 finished with value: 0.8620907581310122 and parameters: {'n_estimators': 453, 'max_depth': 2, 'learning_rate': 0.029134003904067913, 'min_child_weight': 2, 'gamma': 5.639681444120554e-05, 'subsample': 0.2957314874755229, 'colsample_bytree': 0.5673458915456189, 'reg_alpha': 0.08374496394037684, 'reg_lambda': 0.0017056375478477319, 'early_stoppig_rounds': 81}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8620907581310122


[I 2023-01-21 07:22:17,850] Trial 90 finished with value: 0.8531955451213925 and parameters: {'n_estimators': 440, 'max_depth': 3, 'learning_rate': 0.011775738538203054, 'min_child_weight': 4, 'gamma': 0.00013795310115294248, 'subsample': 0.4051113889634612, 'colsample_bytree': 0.5373406670788398, 'reg_alpha': 0.22448396933095133, 'reg_lambda': 0.01454977154679859, 'early_stoppig_rounds': 89}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8531955451213925


[I 2023-01-21 07:22:20,225] Trial 91 finished with value: 0.8655116239120476 and parameters: {'n_estimators': 435, 'max_depth': 2, 'learning_rate': 0.0658583337048027, 'min_child_weight': 3, 'gamma': 0.00028049255095879803, 'subsample': 0.2612165907974111, 'colsample_bytree': 0.6788486710039688, 'reg_alpha': 0.0022619079062986217, 'reg_lambda': 0.06493325633717528, 'early_stoppig_rounds': 40}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8655116239120476


[I 2023-01-21 07:22:21,616] Trial 92 finished with value: 0.8582285272560695 and parameters: {'n_estimators': 221, 'max_depth': 2, 'learning_rate': 0.043250720716579115, 'min_child_weight': 3, 'gamma': 0.00024414447213163615, 'subsample': 0.249947080079865, 'colsample_bytree': 0.39980467145208715, 'reg_alpha': 0.006528506262378707, 'reg_lambda': 0.22301638227027623, 'early_stoppig_rounds': 78}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8582285272560695


[I 2023-01-21 07:22:23,943] Trial 93 finished with value: 0.859819743472286 and parameters: {'n_estimators': 490, 'max_depth': 2, 'learning_rate': 0.050347606602526976, 'min_child_weight': 3, 'gamma': 0.000354453615505063, 'subsample': 0.26645806761773805, 'colsample_bytree': 0.4994963122874942, 'reg_alpha': 0.9932636665631827, 'reg_lambda': 0.38938016072975606, 'early_stoppig_rounds': 54}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.859819743472286


[I 2023-01-21 07:22:25,728] Trial 94 finished with value: 0.8658875400824553 and parameters: {'n_estimators': 422, 'max_depth': 2, 'learning_rate': 0.07636477247880169, 'min_child_weight': 4, 'gamma': 0.008179860029879455, 'subsample': 0.2214968658041648, 'colsample_bytree': 0.7384005908419985, 'reg_alpha': 0.0012923847996631596, 'reg_lambda': 0.0003824176867775201, 'early_stoppig_rounds': 92}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8658875400824553


[I 2023-01-21 07:22:27,638] Trial 95 finished with value: 0.8617726752175905 and parameters: {'n_estimators': 350, 'max_depth': 3, 'learning_rate': 0.053610062187981984, 'min_child_weight': 4, 'gamma': 0.0005509494362732732, 'subsample': 0.3624025171824748, 'colsample_bytree': 0.818097036844489, 'reg_alpha': 0.0008377133950229127, 'reg_lambda': 0.003141792397599696, 'early_stoppig_rounds': 58}. Best is trial 53 with value: 0.8725910444342648.


Avg AUC: 0.8617726752175905


[I 2023-01-21 07:22:28,795] Trial 96 finished with value: 0.872599289967934 and parameters: {'n_estimators': 461, 'max_depth': 2, 'learning_rate': 0.11634909429778409, 'min_child_weight': 1, 'gamma': 0.00036770696205157646, 'subsample': 0.21037865602062264, 'colsample_bytree': 0.607771518151508, 'reg_alpha': 5.279597083207242e-05, 'reg_lambda': 0.5386506165822396, 'early_stoppig_rounds': 43}. Best is trial 96 with value: 0.872599289967934.


Avg AUC: 0.872599289967934


[I 2023-01-21 07:22:30,379] Trial 97 finished with value: 0.8225634448007328 and parameters: {'n_estimators': 475, 'max_depth': 10, 'learning_rate': 0.1695955894770346, 'min_child_weight': 1, 'gamma': 0.0008476317617674433, 'subsample': 0.2099389593064925, 'colsample_bytree': 0.6004776503797347, 'reg_alpha': 6.160979515087835e-05, 'reg_lambda': 0.00844907709037274, 'early_stoppig_rounds': 44}. Best is trial 96 with value: 0.872599289967934.


Avg AUC: 0.8225634448007328


[I 2023-01-21 07:22:31,416] Trial 98 finished with value: 0.8654058634906093 and parameters: {'n_estimators': 337, 'max_depth': 2, 'learning_rate': 0.11640322172399498, 'min_child_weight': 1, 'gamma': 0.005130693116532587, 'subsample': 0.2409058772390851, 'colsample_bytree': 0.7081342777861455, 'reg_alpha': 0.00014354952365015287, 'reg_lambda': 0.19127346965464373, 'early_stoppig_rounds': 42}. Best is trial 96 with value: 0.872599289967934.


Avg AUC: 0.8654058634906093


[I 2023-01-21 07:22:33,327] Trial 99 finished with value: 0.8431724118185983 and parameters: {'n_estimators': 464, 'max_depth': 7, 'learning_rate': 0.09408342271521741, 'min_child_weight': 1, 'gamma': 0.00037365513657391193, 'subsample': 0.20479026479429752, 'colsample_bytree': 0.6243860182113483, 'reg_alpha': 1.9994784604069627e-05, 'reg_lambda': 0.0012711225541654138, 'early_stoppig_rounds': 48}. Best is trial 96 with value: 0.872599289967934.


Avg AUC: 0.8431724118185983


In [33]:
study_xgb.best_value

0.872599289967934

## INSIGHTS:
* With 10 fold cv and target encoding, we achieve a best value of 0.879. But i feel like 10 cv might be too much for such small dataset in that it's way more prone to randoness than a 5 fold is
* With 5 fold cv and target encoding, we achieve a best value of 0.87259 or 0.873

Although the pbL is just luck based casino game at this point, we'll still submit using the best params to make sure we're on the right path with such cross_validation technique. The last time i tried using this technique, it resulted in severe overfitting because idk how i somehow messed something up

In [34]:
study_xgb.best_params

{'n_estimators': 461,
 'max_depth': 2,
 'learning_rate': 0.11634909429778409,
 'min_child_weight': 1,
 'gamma': 0.00036770696205157646,
 'subsample': 0.21037865602062264,
 'colsample_bytree': 0.607771518151508,
 'reg_alpha': 5.279597083207242e-05,
 'reg_lambda': 0.5386506165822396,
 'early_stoppig_rounds': 43}

In [36]:
# # parameters that overfit seeverelyy
# xgb_params = {'n_estimators': 462,
#                  'max_depth': 9,
#                  'learning_rate': 0.24754839357747555,
#                  'min_child_weight': 1,
#                  'gamma': 0.0002768382878439927,
#                  'subsample': 0.4962635986679422,
#                  'colsample_bytree': 0.43112487341982364,
#                  'reg_alpha': 2.4326609397287665e-05,
#                  'reg_lambda': 0.21844284657376126,
#                  'early_stoppig_rounds': 45}

# new shiny params promising to not overfit
xgb_params = {'n_estimators': 461,
                 'max_depth': 2,
                 'learning_rate': 0.11634909429778409,
                 'min_child_weight': 1,
                 'gamma': 0.00036770696205157646,
                 'subsample': 0.21037865602062264,
                 'colsample_bytree': 0.607771518151508,
                 'reg_alpha': 5.279597083207242e-05,
                 'reg_lambda': 0.5386506165822396,
                 'early_stoppig_rounds': 43}

In [37]:
X_train_fr, X_val, y_train_fr, y_val = train_test_split(X_train, y, test_size=0.1, shuffle=True, random_state=1337,
                                                        stratify=y)

In [38]:
xgb_tuned_clf = xgb.XGBClassifier(**xgb_params)
xgb_tuned_clf.fit(X_train_fr, y_train_fr, eval_set=[(X_val, y_val)], verbose=False)

[07:26:28] WARNING: ../src/learner.cc:627: 
Parameters: { "early_stoppig_rounds" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.




XGBClassifier(base_score=0.5, booster='gbtree', callbacks=None,
              colsample_bylevel=1, colsample_bynode=1,
              colsample_bytree=0.607771518151508, early_stoppig_rounds=43,
              early_stopping_rounds=None, enable_categorical=False,
              eval_metric=None, gamma=0.00036770696205157646, gpu_id=-1,
              grow_policy='depthwise', importance_type=None,
              interaction_constraints='', learning_rate=0.11634909429778409,
              max_bin=256, max_cat_to_onehot=4, max_delta_step=0, max_depth=2,
              max_leaves=0, min_child_weight=1, missing=nan,
              monotone_constraints='()', n_estimators=461, n_jobs=0,
              num_parallel_tree=1, predictor='auto', random_state=0,
              reg_alpha=5.279597083207242e-05, ...)

In [39]:
xgb_tuned_preds = xgb_tuned_clf.predict_proba(X_test)[:, 1]

In [41]:
# non-overfitting predictions
submission = pd.DataFrame({"id": test_idx, "Attrition": xgb_tuned_preds})
submission.head()

id  Attrition
0  1677   0.401170
1  1678   0.117693
2  1679   0.000777
3  1680   0.015711
4  1681   0.295577

In [42]:
# OVERFITTED PREDICTIONS
# submission = pd.DataFrame({"id": test_idx, "Attrition": xgb_tuned_preds})
# submission.head()

id  Attrition
0  1677   0.021201
1  1678   0.001126
2  1679   0.000017
3  1680   0.000405
4  1681   0.947035

In [42]:
submission.to_csv("submission.csv", index=False)